datafusion/functions-array/src/kernels.rs

```diff
@@ -319,3 +316,125 @@ pub fn gen_range(
     )?);
     Ok(arr)
 }
+
+/// Returns the length of each array dimension
```

Comment:
I know this is just following the existing pattern of array_functions, but I wonder if it would be better to organize the code by function. 

For example, we could put the UDF and implementations in `datafusion/functions-array/src/dims.rs` 🤔 

We could definitely do this as a follow on PR

Any thoughts @jayzhan211 ?

Reply:
Should we split two files udf and kernel inside `dims`?

Reply:
I think the difficult part is naming the category name. Like `array_append`, `array_prepend` and `array_concat`. Should we name it `mutable`? How about pop_front and pop_back.

Reply:
I agree how to organize the functions is not clear. 

We could always just make one file for each function, I suppose. `array_append.rs`, `array_prepend.rs`, etc. Though that feels like a bit of an overkill